ocw/lib/EC2.py

```diff
@@ -76,6 +77,28 @@ def all_clusters(self):
             [clusters.append(cluster) for cluster in response['clusters']]
         return clusters
 
+    def delete_snapshots_older_than(self, delete_date: datetime.date):
+        regions = self.all_regions()
+        deleted_snapshots = list()
+        for region in regions:
+            response = self.ec2_client(region).describe_snapshots(OwnerIds=['self'])
+            for snapshot in response['Snapshots']:
+                snapshot_start_date = datetime.date(snapshot['StartTime'])
+                if snapshot_start_date < delete_date:
+                    age = date.today() - snapshot_start_date
+                    logger.info("[{}] Deleting snapshot {} in region {} it was {} days old".format(
+                        self.__credentials.namespace, snapshot['SnapshotId'], region, age))
+                    self.ec2_client(region).delete_snapshot(SnapshotId=snapshot['SnapshotId'])
```

Comment:
I guess we need to exception handling here as well

Reply:
I would actually keep it like this for now .
1. We should not compare this flow with delete_instance . Important difference that in case of exception here we don't have important step which we will miss ( update local DB )
2. I am against catch all exceptions blindly . without changing a lot of other things only what we can do in this case add message to log file , which can be easily overlooked
3. I would prefer more approach which I took in https://github.com/cfconrad/pcw/pull/93 where I catch some known exception for which we already face a problem and re-throw everything else
4. worst case scenario which we will get currently - snapshots will be not deleted which was not happening for a quite some time anyway 🙂